awx/main/tasks.py

```diff
@@ -3049,7 +3043,7 @@ def __init__(self, task, runner_params=None):
 
         if self.task and not self.task.instance.is_container_group_task:
             execution_environment_params = self.task.build_execution_environment_params(self.task.instance, runner_params['private_data_dir'])
-            self.runner_params['settings'].update(execution_environment_params)
+            self.runner_params.update(execution_environment_params)
```

Comment:
Is this ok?  We're changing the level in the data structure where we're injecting the EE params.

Reply:
Yeah, a real critical non-obvious thing here is:

https://github.com/ansible/ansible-runner/blob/205c4940e44ae0cdbe418c2b4bcf3d2f312388eb/ansible_runner/config/_base.py#L185

I'm not sure if allowing either was a stroke of genius or complete confusion.

Reply:
Instead of having ansible-runner collect parameters as if they were settings, we're making use of the runner's python API _directly_:
https://github.com/beeankha/ansible-runner/blob/a991e968be08f31182700b0390541ae4e054be5c/ansible_runner/config/_base.py#L66

> I'm not sure if allowing either was a stroke of genius or complete confusion.

I echo this 100%. I guess runner's python API accepts both a `settings` argument as well as the individual settings as kwargs in order to distinguish where they came from?